scripts/services_java.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+function new_service() {
+    # $1 = service type
+    # $2 = port number
+    # $3 = hostname suffix
+    # $4 = additional command line arguments
+    typ=$1
+    port=$2
+    hostname_suffix=$3
+    arguments=$4
+
+    build_script="cd traffic-director-grpc-examples-master/java
+sudo apt-get install -y openjdk-11-jdk-headless
+./gradlew installDist"
+    server="./build/install/wallet/bin/${typ}-server"
+
+    startup_script="#! /bin/bash
+set -ex
+cd /root
+export HOME=/root
+sudo apt-get update -y
+curl -L https://storage.googleapis.com/traffic-director/td-grpc-bootstrap-0.9.0.tar.gz | tar -xz
+./td-grpc-bootstrap-0.9.0/td-grpc-bootstrap | tee /root/td-grpc-bootstrap.json
+curl -L https://github.com/GoogleCloudPlatform/traffic-director-grpc-examples/archive/master.tar.gz | tar -xz
+${build_script}
+sudo systemd-run -E GRPC_XDS_BOOTSTRAP=/root/td-grpc-bootstrap.json ${server} --port=${port} --hostname_suffix=${hostname_suffix} ${arguments}"
+
+    gcloud compute instance-templates create grpcwallet-${hostname_suffix}-template \
+      --scopes=https://www.googleapis.com/auth/cloud-platform \
+      --tags=allow-health-checks \
+      --image-family=debian-10 \
+      --image-project=debian-cloud \
+      --metadata-from-file=startup-script=<(echo "${startup_script}")
```

Comment:
Is this the best way to do this?  What if the startup script was checked in as its own file, as a template, and this line fills in the right values?

Reply:
The template being a file containing `startup_script`, but with placeholders where the variables are expanded into.  But if you need to have only one file contain everything then ignore this.